### PR TITLE
Replace var by let and const (Fix #15), Fix #17

### LINKED ISSUE #15: Replace var by let
let are better scoped than var, etc. so except for backward compatibility it has no reason to use var rather than let

### LINKED ISSUE #17: Replace string concatenation by template literals
In your code, sometimes you use template literals, sometimes you used string concatenation. It's better to use template literals when working with variables, as it's more readable.

### PR DESCRIPTION
**Pull request by AntoineJT**
Just solve the issue I opened some days ago, by replacing var by let and const and so patching some bad variable scoping.

### CODE REVIEW

File: js/app.js [hunk 1]
```diff
@@ -1,39 +1,44 @@
...
             return "hsl("+ set[i][0] +","+ set[i][1] +"%,"+ set[i][2] +"%)";
         }
         function modHSL(str){
-        var arr = str.replace(/[^\d,.%]/g, '').split(',').map(x => Number.parseFloat(x, 10));
-        return `hsl(${arr[0]-25},${arr[1]-10}%,${arr[2]-7}%)`;
+            const arr = str.replace(/[^\d,.%]/g, '').split(',').map(x => Number.parseFloat(x, 10));
+            return `hsl(${arr[0]-25},${arr[1]-10}%,${arr[2]-7}%)`;
         }
         function rota(arr, mode, w, h) {
+            let arr2 = [];
```
[the remark this answers is not in the document]
Comment (AntoineJT) [reply js/app.js]: This is just a forbidden change

File: js/app.js [hunk 2]
```diff
@@ -43,41 +48,41 @@
...
-        for(var word = r[1], i = 0; i < 25; i++) {
-            var bit = word & 1;
+        for(let word = r[1], i = 0; i < 25; i++) {
+            const bit = word & 1;
             a.push(bit); word >>>= 1;
         }
-        for(var word = r[0], i = 0; i < 25; i++) {
-            var bit = word & 1;
+        for(let word = r[0], i = 0; i < 25; i++) {
+            const bit = word & 1;
```
[the remark this answers is not in the document]
Comment (bryc) [reply js/app.js]: Fixed `bit` in diff sent you.

File: js/app.js [hunk 3]
```diff
@@ -119,14 +125,14 @@
...
-        for(var o = t.width/n, Q=d=y=s= 0; s < l; s++, d = s%(n/2)) {
+        for(let o = t.width/n, Q=d=y=s= 0; s < l; s++, d = s%(n/2)) {
             // Change color at halfway point. NOTE: |0 required for odd sizes.
             (s === (0|l/2)) && (c.fillStyle = color3, Q += q/2, y =- 1);
             (s && !d) && y++; // Increment y axis.
             (a[s]) && c.fillRect(Q+o*d, o*y, o, o); // If pixel exists, fill square on canvas (x, y, w, h).
         }
         c.fillStyle = color2;
-        for(var o = t.width/n, Q=d=y=s= 0; s < l; s++, d = s%(n/2)) {
+        for(let o = t.width/n, Q=d=y=s= 0; s < l; s++, d = s%(n/2)) {
```
[the remark this answers is not in the document]
Comment (bryc) [reply js/app.js]: I rewrote this with proper usage of `const` and `let` in diff.

File: js/app.js [hunk 4]
```diff
@@ -1,39 +1,44 @@
...
             }
             if(mode===1||mode===4) arr2.reverse();
             return arr2;
         }
-        var n = 10, q = n*3, l = n*n, a = [], c = t.getContext("2d");
+        const n = 10;
+        const q = n*3;
+        const l = n*n;
+        let a = [];
+        const c = t.getContext("2d");
```
[the remark this answers is not in the document]
Comment (AntoineJT) [reply js/app.js]: No, ``b`` must be declared just before it's used, else it's a pain for readability. Move ``b`` up is not clearer at all.
It has 2 options : 
- declare all variables and constants at the very beginning of the function, and then use them later (not really readable if it has alot I think);
- declare variables and constants just before they're used (which is really readable I think).

**Pull request merged**